### PR TITLE
Remove ComboColour from HitObjects

### PR DESCRIPTION
**Pull request by peppy**
Prerequisite step to having combo colours provided by external sources (skins, beatmap).

Prepares `AccentColour` setters to correctly update `DrawableHitObject` pieces.

Note that merging this will remove combo colour assignment temporarily.

### CODE REVIEW

File: osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        /// <summary>
+        /// The offset of this hitobject in the current combo.
+        /// </summary>
+        int IndexInCurrentCombo { get; set; }
+
+        /// <summary>
+        /// The offset of this hitobject in the current combo.
+        /// </summary>
+        int ComboIndex { get; set; }
```
Comment (smoogipoo) [inline osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs]: Why is this property duplicated?

Comment (smoogipoo) [reply osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs]: If this is the index of the combo as a whole in the beatmap, this should be in `IHasCombo`

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs]: hmm, I split these out based on usage. basically there are many places we mark objects as supporting combo, but don't require any knowledge beyond the `bool NewCombo`.

would renaming this class to `IHasComboInformation` help?

Comment (smoogipoo) [reply osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs]: I don't see any situation where you'd want to know only of `NewCombo` and nothing else. There's not enough data there to tell you anything.

Example?

**Comment by smoogipoo** [comment]
half-approving this one, until I check further PRs

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**